### PR TITLE
workspace vitest config

### PR DESCRIPTION
**Pull request by turbocrime**
vitest updated and added some nice features for monorepo/workspace tests

`pnpm test` at the root uses turbo to call the test script of each package, which is just `vitest run`

`pnpm vitest` from the workspace root or in any package will watch tests with the appropriate config, and `pnpm vitest run` will one-shot them with the appropriate config

CI workflow is updated to no longer matrix, as test configs and environments now resolve properly to each test file without command-line arguments.

the vscode vitest extension now supports workspaces, and this config, so those of us running vscode get a nice watch ui with jump-to-failures and a debugger

Screenshot 2024-02-28 at 14 44 26

### CODE REVIEW

File: packages/ui/lib/toast/toast.test.ts
Comment (turbocrime) [inline packages/ui/lib/toast/toast.test.ts:240]: updated these dynamically generated tests to just iterate within the test, because the vscode extension had trouble understanding what was happening here

File: turbo.json
Comment (turbocrime) [inline turbo.json:13]: also cacheing download properly now

File: apps/extension/vitest.config.ts
Comment (turbocrime) [inline apps/extension/vitest.config.ts:1]: extension doesn't use vite, just vitest

File: packages/router/vitest.config.ts
Comment (turbocrime) [inline packages/router/vitest.config.ts:1]: router doesn't use vite, just vitest

File: packages/storage/src/chrome/migration.test.ts
Comment (grod220) [inline packages/storage/src/chrome/migration.test.ts:149]: If we remove `flushPromises` we've lost some documentation. Can you add that here?

File: packages/crypto/package.json
Comment (grod220) [inline packages/crypto/package.json:8]: How do these tests know to run in a browser? Way back we had issues with the global `crypto` package being used in a nodejs environment.

**Review by grod220 — APPROVED** [review]

File: packages/storage/src/chrome/migration.test.ts
Comment (grod220) [reply packages/storage/src/chrome/migration.test.ts:149]: Looks like it was insta merged, but could we still address this one ⬆️

**Comment by grod220** [comment]
Come to think of it, maybe we shouldn't have auto-merge 🤔. It makes sense after an approval (so you don't have to wait on the build), but before an approval it's possibly too hasty.

File: packages/storage/src/chrome/migration.test.ts
Comment (turbocrime) [reply packages/storage/src/chrome/migration.test.ts:149]: it seems unnecessary so i'll just make a PR that removes it.

File: packages/crypto/package.json
Comment (turbocrime) [reply packages/crypto/package.json:8]: package-level config runs it in playwright

https://github.com/penumbra-zone/web/blob/ea1c3740e030a29705cdf3ac5e4dbd0c15e6cefe/packages/storage/vitest.config.ts#L1-L14

**Comment by turbocrime** [comment]
ew, it cluttered up the commit log too. i expected a squash

**Pull request merged**
